bndtools.core/src/bndtools/explorer/BndtoolsExplorer.java

```diff
@@ -52,66 +88,23 @@ public void createPartControl(Composite parent) {
 		}
 		c.layout();
 
-
 		filterPart.setHint("Filter for projects (glob)");
-		listener = e -> {
-			String value = (String) e.getNewValue();
-
-			if (Strings.nonNullOrEmpty(value)) {
-				glob = new Glob(value);
-				if (!installed) {
-					installed = true;
-					getTreeViewer().addFilter(new ViewerFilter() {
-
-						@Override
-						public boolean select(Viewer viewer, Object parentElement, Object element) {
-							if (glob == null)
-								return true;
-
-							if (element instanceof JavaProject) {
-								IJavaProject project = (JavaProject) element;
-								String name = project.getElementName();
-								return isSelected(project.getProject(), name);
-
-							} else if (element instanceof IProject) {
-								IProject project = (IProject) element;
-								String name = project.getName();
-								return isSelected(project, name);
-							} else
-								return true;
-						}
-
-						private boolean isSelected(IProject project, String name) {
-							if (glob.finds(name) >= 0)
-								return true;
-
-							try {
-								int maxSeverity = project.findMaxProblemSeverity(null, false, IResource.DEPTH_INFINITE);
-
-								switch (maxSeverity) {
-									case 0 :
-										return false;
-									case IMarker.SEVERITY_ERROR :
-										return glob.finds(":error") >= 0;
-
-									case IMarker.SEVERITY_WARNING :
-										return glob.finds(":warning") >= 0;
-								}
-							} catch (CoreException e) {
-								// ignore
-							}
-
-							return false;
-
-						}
-					});
-				}
-			} else {
-				glob = null;
-			}
-			getTreeViewer().refresh();
+		filterListener = e -> {
+			Glob old = glob;
+			glob = null;
+			String filterText = (String) e.getNewValue();
+			if (Strings.nonNullOrEmpty(filterText))
+				glob = new Glob(filterText);
+			if (!Objects.equals(old, glob))
+				update();
 		};
-		filterPart.addPropertyChangeListener(listener);
+		filterPart.addPropertyChangeListener(filterListener);
+
+		IWorkbenchWindow activeWorkbenchWindow = workbench.getActiveWorkbenchWindow();
```

Comment:
```suggestion
Optional.ofNullable(workbench.getActiveWorkbenchWindow())
			.map(IWorkbenchWindow::getActivePage)
			.ifPresent(page -> page.addPartListener(partListener));
```

Reply:
I thought about this but this is a part that is being constructed. How can we construct a part when there is no activeWorkbenchWindow? I am assuming it is never null at that place?

Reply:
It is possible to have no active workbench window but it's not worth
considering IMHO :)